src/Microsoft.IdentityModel.Tokens/JsonWebKey.cs

```diff
@@ -25,18 +25,18 @@
 //
 //------------------------------------------------------------------------------
 
-using Microsoft.IdentityModel.Logging;
-using Newtonsoft.Json;
 using System;
 using System.Collections.Generic;
+using Microsoft.IdentityModel.Logging;
+using Newtonsoft.Json;
 
-namespace System.IdentityModel.Tokens.Jwt
+namespace Microsoft.IdentityModel.Tokens
 {
     /// <summary>
     /// Represents a Json Web Key as defined in http://tools.ietf.org/html/rfc7517.
     /// </summary>
     [JsonObject]
-    public class JsonWebKey
+    public class JsonWebKey : AsymmetricSecurityKey
```

Comment:
To enable creating signature providers directly from JsonWebKey. Currently, we have `GetSigningKeys` method on JsonWebKeySet that translates a JWK to a `SecurityKey`. But that is not complete and we are considering getting rid of it. Making JWK inherit from `SecurityKey` will allow creating signature providers from these keys directly. Also, this enables users to obtain parameters from metadata endpoint and hook in their own signature provider (one that works on unix) by overriding `GetSignatureProvider`.